node/node.go

```diff
@@ -145,7 +149,24 @@ func (n *Node) Run(ctx context.Context) {
 
 	http := makeHTTP(n.cfg.RPCPort, rpc.New(n.blockchain, n.cfg.Network), n.log)
 
-	n.services = []service.Service{synchronizer, http}
+	ethClient, err := ethclient.Dial(n.cfg.EthNode)
+	if err != nil {
+		n.log.Errorw("Cannot connect to Ethereum node", "err", err)
+		return
+	}
+	coreContractAddress, err := n.cfg.Network.CoreContractAddress()
+	if err != nil {
+		n.log.Errorw("Error finding core contract address for network", "err", err, "network", n.cfg.Network)
+		return
+	}
+	coreContractFilterer, err := contract.NewStarknetFilterer(*coreContractAddress, ethClient)
+	if err != nil {
+		n.log.Errorw("Failed to set up filterer for L1 contract", "err", err, "network", n.cfg.Network, "ethereumNode", n.cfg.EthNode)
+		return
+	}
+	l1Client := l1.NewClient(l1.NewWatcher(ethClient, coreContractFilterer), n.blockchain, n.log)
```

Comment:
I think we can move all this to l1.go
It might make testing a little harder but we can always introduce a helper to make the dependency injection work.